src/main/java/org/example/movieSpring/service/MovieService.java

```diff
@@ -0,0 +1,51 @@
+package org.example.movieSpring.service;
+
+import org.example.movieSpring.model.Movie;
+import org.example.movieSpring.repo.MovieRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class MovieService {
+
+    private final MovieRepo movieRepo;
+
+    @Autowired
+    public MovieService(MovieRepo movieRepo) {
+        this.movieRepo = movieRepo;
+    }
+
+    public List<Movie> getMovies() {
+        return movieRepo.findAll();
+    }
+
+    public Optional<Movie> getMovieById(Long id) {
+        return movieRepo.findById(id);
+    }
+
+    public Movie addMovie(Movie movie) {
+        return movieRepo.save(movie);
+    }
+
+    public List<Movie> getMoviesByReleaseYear(int releaseYear) {
+        return movieRepo.findAll().stream().filter(m -> m.getRelease_year() == releaseYear).toList();
+    }
```

Comment:
this should be done at the database level, databases are specialised software to handle all kinds of operations on data. often they come with specialised hardware too. one of these operations is filtering, this should be delegated to the db

Reply:
nvm fixed it